apps/builder/app/canvas/shared/styles.ts

```diff
@@ -191,19 +191,16 @@ const getOrCreateRule = ({
   const key = `${instanceId}:${breakpointId}:${state}`;
   let rule = wrappedRulesMap.get(key);
   if (rule === undefined) {
-    const params = getParams();
-    rule = cssEngine.addStyleRule(
-      `[${idAttribute}="${instanceId}"]${state}`,
-      {
-        breakpoint: breakpointId,
-        style: {},
-      },
-      createImageValueTransformer(assets, {
-        assetBaseUrl: params.assetBaseUrl,
-      })
-    );
+    rule = cssEngine.addStyleRule(`[${idAttribute}="${instanceId}"]${state}`, {
+      breakpoint: breakpointId,
+      style: {},
+    });
     wrappedRulesMap.set(key, rule);
   }
+  const params = getParams();
+  rule.styleMap.setTransformer(
+    createImageValueTransformer(assets, { assetBaseUrl: params.assetBaseUrl })
+  );
```

Comment:
we need to start adding tests to these functions in styles.ts, they are super mission critical and completely untested